### ISSUE TITLE
Search menu styling + a few questions 

### ISSUE THREAD

**Issue by Haitch-1** [ramensoftware/windows-11-start-menu-styling-guide#14]
Is it possible to modify the search menu aswell? (SearchHost.exe)
I couldn't get it to work.

And for my other question: How can I make the start menu bigger without breaking anything?

**Comment by bbmaster123** [comment]
yes its possible if you also inject the start menu mod into searchhost, and works quite well already, but I'm not sure if support will be added to the start menu mod, or if it will end up being a separate mod. Also not sure about bugs, although I haven't found any yet myself

the only way I've found so far is to use Transform3d 
`Transform3D:=<CompositeTransform3D CenterX="275" CenterX="275" ScaleX="1.965%" ScaleY="1.965%" ScaleZ="15%"TranslateY="2" />`

there is also a scale property, and also TransformMatrix, but I haven't looked into those yet

CenterX defines the center of the width of the menu, so if the menu is currently 500 pixels wide, put in 250
CenterY is the same but for height
ScaleX and ScaleY define how much to squish or stretch the menu starting from the center point defined
ScaleZ defines how far or close in 3d space to move the menu. Say if you want a button to pop up or press in when clicked, for example
TranslateX, Y and Z move the menu in the axis specified without changing its size

PS: search isn't the only thing that ALSO works :)

**Comment by m417z** [comment]
Note that while it's possible, most of it is actually a web view control, so customizing the content will require a different approach.

See also: https://github.com/m417z/UWPSpy/issues/3

**Comment by MuscularPuky** [comment]
have you try this?![image](https://github.com/ramensoftware/windows-11-start-menu-styling-guide/assets/93962018/7debbe54-7f27-4071-9d2f-06cc07665c83)

**Comment by LucasPC-hub** [comment]
> yes its possible if you also inject the start menu mod into searchhost, and works quite well already, but I'm not sure if support will be added to the start menu mod, or if it will end up being a separate mod. Also not sure about bugs, although I haven't found any yet myself
> 
> the only way I've found so far is to use Transform3d `Transform3D:=<CompositeTransform3D CenterX="275" CenterX="275" ScaleX="1.965%" ScaleY="1.965%" ScaleZ="15%"TranslateY="2" />`
> 
> there is also a scale property, and also TransformMatrix, but I haven't looked into those yet
> 
> CenterX defines the center of the width of the menu, so if the menu is currently 500 pixels wide, put in 250 CenterY is the same but for height ScaleX and ScaleY define how much to squish or stretch the menu starting from the center point defined ScaleZ defines how far or close in 3d space to move the menu. Say if you want a button to pop up or press in when clicked, for example TranslateX, Y and Z move the menu in the axis specified without changing its size
> 
> PS: search isn't the only thing that ALSO works :)

where do i put this command ? do put it in "value" after variable key? if it is,its not working for me

**Comment by LucasPC-hub** [comment]
> yes its possible if you also inject the start menu mod into searchhost, and works quite well already, but I'm not sure if support will be added to the start menu mod, or if it will end up being a separate mod. Also not sure about bugs, although I haven't found any yet myself
> 
> the only way I've found so far is to use Transform3d `Transform3D:=<CompositeTransform3D CenterX="275" CenterX="275" ScaleX="1.965%" ScaleY="1.965%" ScaleZ="15%"TranslateY="2" />`
> 
> there is also a scale property, and also TransformMatrix, but I haven't looked into those yet
> 
> CenterX defines the center of the width of the menu, so if the menu is currently 500 pixels wide, put in 250 CenterY is the same but for height ScaleX and ScaleY define how much to squish or stretch the menu starting from the center point defined ScaleZ defines how far or close in 3d space to move the menu. Say if you want a button to pop up or press in when clicked, for example TranslateX, Y and Z move the menu in the axis specified without changing its size
> 
> PS: search isn't the only thing that ALSO works :)

hey man can you please tell me what to target?

**Comment by bbmaster123** [comment]
@LucasPC-hub I'd rather teach you [how to target](https://github.com/bbmaster123/FWFU/blob/main/uwpspy.md)
This guide isn't quite finished yet (needs more images) but it should help teach the process. 

give that a quick read and if there is anything that you don't fully understand (or is missing in the guide) let me know and I'll send over the correct target for search. 
cheers :)

**Comment by TheMasterOfMadness** [comment]
So I am using the Fluent2Inspired Theme and I tried figuring out to get the Search Menu to be themed, but it didn't work.
I have added searchhost.exe to the inclusion list as I'm on Win11 24h2
This is how it still looks:
![image](https://github.com/user-attachments/assets/5fded46e-a875-4477-b675-88d8dc635f51)

So am I doing something wrong or missing something?

**Comment by MuscularPuky** [comment]
> So I am using the Fluent2Inspired Theme and I tried figuring out to get the Search Menu to be themed, but it didn't work. I have added searchhost.exe to the inclusion list as I'm on Win11 24h2 This is how it still looks: ![image](https://private-user-images.githubusercontent.com/39934989/392857128-5fded46e-a875-4477-b675-88d8dc635f51.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0NTE4MDYsIm5iZiI6MTczMzQ1MTUwNiwicGF0aCI6Ii8zOTkzNDk4OS8zOTI4NTcxMjgtNWZkZWQ0NmUtYTg3NS00NDc3LWI2NzUtODhkOGRjNjM1ZjUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA2VDAyMTgyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4ZWZmNTZlOGNhZTE2ZTFiOGM3Y2U1MjBjOGRjZWRmYjYwNTNkNmZmODg2Yjk3NDkyMDI3ZjY3ZmVkNzY0YjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.l7k87zoER6rls365eJF1OVmjqrJRhooJUGbYCMKBfEY)
> 
> So am I doing something wrong or missing something?

I think there;s some search acrylic layer change in 24H2. suddenly my text rendering in search pane is become fine and clear

**Comment by SandTechStuff** [comment]
@TheMasterOfMadness Try using `searchexperiencehost.exe`.

@bbmaster123 I am on 23H2 and Search is running under `searchhost.exe`, not `searchexperiencehost.exe`. Maybe you got it mixed up in the readme update? TheMasterOfMadness' issue supports this, as `searchhost.exe` did not work for him, so maybe `searchexperiencehost.exe` is for 24H2.

**Comment by TheMasterOfMadness** [comment]
I have both of those files included so they work on the Translucent one but
not on the Fluent one.

**Comment by m417z** [comment]
> I am on 23H2 and Search is running under `searchhost.exe`, not `searchexperiencehost.exe`, just something to note as the readme says otherwise

I guess it's a confusion from `ShellHost.exe` (appeared in 24H2) and `ShellExperienceHost.exe`. I removed `searchexperiencehost.exe` from the readme.
